tests/System.Binary.Base64.Tests/PerformanceTests.cs

```diff
@@ -96,5 +119,176 @@ private static void Base64DecodeBaseline(int numberOfBytes)
+                            if (leftOverBytes < 4)
+                            {
+                                source1.Slice(bytesConsumed).CopyTo(stackSpan);
+                                int amountToCopy = Math.Min(source2.Length, stackSpan.Length - leftOverBytes);
+                                source2.Slice(0, amountToCopy).CopyTo(stackSpan.Slice(leftOverBytes));
+
+                                Base64Encoder.TryDecode(stackSpan, destination.Slice(bytesWritten), out int consumed, out int written);
+                                afterMergeSlice = consumed - leftOverBytes;
+                                bytesConsumed += consumed;
+                                bytesWritten += written;
+                            }
+                        }
+                        Base64Encoder.TryDecode(source2.Slice(afterMergeSlice), destination.Slice(bytesWritten), out int consumed3, out int written3);
+                        bytesConsumed += consumed3;
+                        bytesWritten += written3;
+                    }
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [InlineData(600, 1000)]
+        [InlineData(3000, 5000)]
+        [InlineData(6000, 10000)]
+        [InlineData(12000, 20000)]
+        [InlineData(30000, 50000)]
+        private static void StichingTestNoThirdCall(int stackSize, int inputBufferSize)
+        {
+            Span<byte> source = new byte[inputBufferSize];
+            InitalizeDecodableBytes(source);
+
+            int misalignedBoundary = inputBufferSize / 5 * 2 + 2;  // 1000 -> 402
+            ReadOnlySpan<byte> source1 = source.Slice(0, misalignedBoundary);
+            ReadOnlySpan<byte> source2 = source.Slice(misalignedBoundary, inputBufferSize - misalignedBoundary);
+
+            Span<byte> stackSpan;
+
+            unsafe
+            {
+                byte* stackBytes = stackalloc byte[stackSize];
+                stackSpan = new Span<byte>(stackBytes, stackSize);
+            }
+
+            Span<byte> destination = new byte[inputBufferSize]; // Plenty of space
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        if (!Base64Encoder.TryDecode(source1, destination, out int bytesConsumed, out int bytesWritten))
+                        {
+                            int leftOverBytes = source1.Length - bytesConsumed;
+                            if (leftOverBytes < 4)
+                            {
+                                source1.Slice(bytesConsumed).CopyTo(stackSpan);
+                                int amountToCopy = Math.Min(source2.Length, stackSpan.Length - leftOverBytes);
+                                source2.Slice(0, amountToCopy).CopyTo(stackSpan.Slice(leftOverBytes));
```

Comment:
The intent was to just measure the cost of stitching compared to a baseline. I will have better perf tests that measure the full piping loop.